toBeArrayOfObjectWith.js

```diff
@@ -0,0 +1,29 @@
+expect.extend({
+  toBeArrayOfObjectWith(received, ...keys) {
+    const hasValidProperties = (object) => keys.reduce(
+      (
+        accumulator, current,
+      ) => accumulator && Object.prototype.hasOwnProperty.call(object, current),
+      true,
+    );
+
+    const failingElement = received.reduce((accumulator, current) => {
+      if (!hasValidProperties(current)) {
+        return current;
+      }
+
+      return accumulator;
+    }, null);
+
+    return (failingElement === null)
+      ? {
+        pass: true,
+        message: () => (`expected ${this.utils.printReceived(failingElement)} to not has ${this.utils.printExpected(keys.join(', '))} as property`),
+      }
+
+      : {
+        pass: false,
+        message: () => (`expected ${this.utils.printReceived(failingElement)} to has ${this.utils.printExpected(keys.join(', '))} as property`),
+      };
```

Comment:
```suggestion
    return {
        pass: failingElement === null,
        message: () => (`expected ${this.utils.printReceived(failingElement)} to ${failingElement === null ? 'not' : ''} has ${this.utils.printExpected(keys.join(', '))} as property`),
      };
```

비슷한 객체인 것 같아 간단하게 수정해봤어요. 몇몇 부분에서만 예외처리가 필요하다면 이런식으로 처리해주는 것도 좋을 것 같네요

Reply:
```js
    const { printReceived, printExpected } = this.utils;

    return {
      pass: failingElement === null,
      message: () => (
        `expected ${printReceived(failingElement)} to 
        ${failingElement === null ? 'not' : ''} has 
        ${printExpected(keys.join(', '))} as property`
      ),
    };
```

중복을 제거할 수 있었군요. 위와 같이 수정해줬습니다.